src/EFRepository.Generator/ExtensionMethodGenerator.cs

```diff
@@ -345,36 +365,102 @@ public static partial class {dbSetClass.Name}Extensions
 			}
 		}
 
-		// Close Class
+		// Close Class & Namespace
 		builder.AppendLine("	}")
 			.AppendLine("}");
 
 		return (fileName, builder.ToString());
 
 		string CreateMethod(string type, string memberName, bool nullable)
 		{
-			string result = $@"
+			StringBuilder result = new();
+			result.AppendLine($@"
 		/// <summary>
 		/// Filter the <see cref=""IQueryable""/> of {dbSetClass.Name} by {memberName}
 		/// </summary>
 		/// <param name=""value"">The {type} which should equal {memberName}</param>
+		/// <exception cref=""ArgumentNullException"">thows if query is null</exception>
 		public static IQueryable<{dbSetClass.Name}> By{memberName}(this IQueryable<{dbSetClass.Name}> query, {type}? value)
 		{{
-			if (query == null)
-				return query;
+			if (query == null) throw new ArgumentNullException(nameof(query));
 
 			if (value == null)
 				return query;
 
 			return query.Where(n => n.{memberName} == value);
-		}}";
+		}}");
+
+			// This is for numeric types only
+			if (!type.Contains("bool") && !type.Contains("DateTime") && !type.Contains("string"))
+			{
+				result.AppendLine($@"
+		/// <summary>
+		/// Filter the <see cref=""IQueryable""/> of {dbSetClass.Name} by {memberName} greater than value
+		/// </summary>
+		/// <param name=""value"">The {type} which should be greater than {memberName}</param>
+		/// <exception cref=""ArgumentNullException"">thows if query is null</exception>
+		public static IQueryable<{dbSetClass.Name}> By{memberName}GreaterThan(this IQueryable<{dbSetClass.Name}> query, {type}? value)
+		{{
+			if (query == null) throw new ArgumentNullException(nameof(query));
+
+			if (value == null)
+				return query;
+
+			return query.Where(n => n.{memberName} > value);
+		}}
+
+		/// <summary>
+		/// Filter the <see cref=""IQueryable""/> of {dbSetClass.Name} by {memberName} greater than or equal value
+		/// </summary>
+		/// <param name=""value"">The {type} which should be greater than {memberName}</param>

```

Comment:
Minor complaint, but this comment should say "greater than or equal to {memberName}"